### PR TITLE
Improve contributing guide and workflow

### PR DESCRIPTION
**Pull request by MaximeKjaer**
This PR:

- Fixes the contributing guide on the topic of the build artifact (see #23),
- Moves all build commands to be NPM / Yarn scripts defined in `package.json`,
- Removes an unused devDependency

This should simplify the workflow little, as there is a Yarn task for every possible action. The basic workflow now only consists of `yarn install`, `yarn build` and `yarn test`. But you can still run very more precise tasks through `yarn test:unit` or `yarn build:extension`, for instance.

### CODE REVIEW

File: bin/ci-release.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ NEW_VERSION=${TRAVIS_TAG#"v"}
 
 # Build the extension
 yarn install
-yarn build
+yarn build:syntax
```
Comment (nicolasstucki) [inline bin/ci-release.sh:9]: When is `yarn build:extension` executed? Is that part of `yarn vscode:publish`?

Comment (MaximeKjaer) [reply bin/ci-release.sh:9]: Yes it is. From the [vsce docs](https://code.visualstudio.com/api/working-with-extensions/continuous-integration#travis-ci-automated-publishing) it seems that `yarn vscode:publish` (aka `vsce publish`) does the packaging before publishing, as it is the only command needed to publish.

File: package.json [hunk 2]
```diff
@@ -44,16 +44,20 @@
...
         "ajv": "^6.12.2",
+        "npm-run-all": "^4.1.5",
         "ts-node": "^8.10.2",
-        "tsc": "^1.20150623.0",
         "typescript": "^3.9.5",
         "vsce": "^1.77.0",
         "vscode-tmgrammar-test": "0.0.10"
     },
     "scripts": {
         "vscode:prepublish": "test -f ./syntaxes/Scala.tmLanguage.json",
```
Comment (MaximeKjaer) [inline package.json:54]: I don't think this `prepublish` script is actually doing anything useful? I tried messing with the json file but this doesn't result in failures of `vscode:prepublish`

Comment (PanAeon) [reply package.json:54]: @MaximeKjaer , it  just checks if the file `./syntaxes/Scala.tmLanguage.json` exists.

File: bin/ci-release.sh (hunk 1 above)
Comment (nicolasstucki) [reply bin/ci-release.sh:9]: Ok

**Pull request merged**